tools/prepare_aa.cwl

```diff
@@ -10,18 +10,36 @@ requirements:
   - class: ResourceRequirement
     ramMin: 16000
     coresMin: $(inputs.threads)
-baseCommand: [mkdir, data_repo]
+  - class: InitialWorkDirRequirement
+    listing: [$(inputs.data_repo)]
+
+baseCommand: [export]
 arguments: 
-  - position: 1
-    shellQuote: false
-    valueFrom: >- 
-      && tar -C data_repo -xzf
-  - position: 2
+  - position: 0
     shellQuote: false
     valueFrom: >-
-      && export AA_DATA_REPO=$PWD/data_repo
+      AA_DATA_REPO=$(inputs.data_repo.path)
       && export AA_SRC=/home/programs/AmpliconArchitect-master/src
       && touch $AA_DATA_REPO/coverage.stats
+      ${
+        if (inputs.mosek_license_file){
+        var cmd = "&& mkdir licenses && cp " + inputs.mosek_license_file.path + " licenses\
+        && export MOSEKLM_LICENSE_FILE=$PWD/licenses";
+        return cmd;
+        }
+        else{
+          return ""
+        }
+      }
+      ${
+        if (inputs.run_ac){
+        var cmd = "&& export AC_SRC=/home/programs/AmpliconClassifier-main";
+        return cmd;
+        }
+        else{
```

Comment:
Initially I was finding some of the more complex valueFroms hard to follow. I wonder if creating a bash wrapper at runtime could be more straightforward, as shown here: https://www.commonwl.org/user_guide/14-runtime/index.html

Reply:
A good suggestion. Especially for complex steps that use pipefail and a lot of different tools.

Reply:
Interesting thought, so, more like, when I have these java script conditionals, create a shell script at the beginning that will be run if the conditional is met? Like for instance: 
```yaml
      ${
        if (inputs.mosek_license_file){
          var cmd = "&& mkdir licenses && cp " + inputs.mosek_license_file.path + " licenses\
          && export MOSEKLM_LICENSE_FILE=$PWD/licenses";
          return cmd;
        }
        else{
          return ""
        }
      }
```

becomes:
```yaml
    listing:
      - entryname: cp_license.sh
        entry: |-
          mkdir license
          cp $(inputs.mosek_license_file.path licenses\
          export MOSEKLM_LICENSE_FILE=\$PWD/licenses
```
as the script, and the js conditional becomes:
```js
      ${
        if (inputs.mosek_license_file){
          return " && bash cp_license.sh";
      }
     else{
      return "";
     }
```
Something like that? Or also replace all the conditionals/section with a big shell script?

Reply:
My thought was the latter, to include the conditionals in the shell script

Reply:
Ok, it  was a little tricky, but I updated the three tools that had exported and replaced them with shell s scripts that get sourced. Downside is if the script fails, not sure how well the error is caught